doc/hacking.rst

```diff
@@ -779,46 +779,115 @@ https://ocaml.github.io/dune/dev/bench/
 Monorepo Benchmark
 ------------------
 
-The file bench/monorepo/bench.Dockerfile sets up a Docker container for
-benchmarking Dune building a large monorepo constructed with
-`opam-monorepo <https://github.com/tarides/opam-monorepo>`_. The monorepo is
-constructed according to the files in
-https://github.com/ocaml-dune/ocaml-monorepo-benchmark/tree/main/benchmark.
-Build the Docker image from the root directory of this repo.
+The Monorepo Benchmark is designed to assess the performance of Dune in
+building a large monorepo, specifically constructed using `opam-monorepo
+<https://github.com/tarides/opam-monorepo>`_.
 
-E.g., run
+Current Monorepo Benchmarks
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-.. code:: console
+TODO
 
-   $ docker build . -f bench/monorepo/bench.Dockerfile --tag=dune-monorepo-benchmark
+Adding new benchmarks
+^^^^^^^^^^^^^^^^^^^^^
 
-The monorepo benchmark Docker image requires ``duniverse`` directory to be mounted
-as a volume. Generate this directory with a script from the
-`ocaml-monorepo-benchmark <https://github.com/ocaml-dune/ocaml-monorepo-benchmark>`_
-repository:
+TODO
 
-.. code:: console
+Setting Up the Benchmark Environment
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-   $ git clone https://github.com/ocaml-dune/ocaml-monorepo-benchmark.git
-   $ cd ocaml-monorepo-benchmark
-   $ ./generate-duniverse.sh /tmp
+To set up the Docker environment for the Monorepo Benchmark, you will need to
+build a Docker image and generate and mount a ``duniverse`` directory.
 
-This will create a directory ``/tmp/duniverse``. Then to run the benchmark, run
-the Docker image in a container mounting ``/tmp/duniverse`` as a volume at
-``/home/opam/bench-dir/current-bench-data/duniverse`` (that specific path is a
-requirement of `current-bench <https://github.com/ocurrent/current-bench>`_).
-From within the container the benchmarks can be started by running `make bench`.
-Do all this in a single command with:
+Building the Docker Image
+^^^^^^^^^^^^^^^^^^^^^^^^^
 
-.. code:: console
+#. **Build the Docker Image**: From the root directory of your local Dune
+   repository, build the Docker image using the
+   ``bench/monorepo/bench.Dockerfile``
+
+   .. code:: console
+
+      $ docker build . -f bench/monorepo/bench.Dockerfile --tag=dune-monorepo-benchmark
+
+Preparing the ``duniverse`` Directory
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+#. **Clone the Benchmark Repository**: Clone the ``ocaml-monorepo-benchmark``
+   repository to obtain necessary scripts and files.
+
+   .. code:: console
+
+      $ git clone https://github.com/ocaml-dune/ocaml-monorepo-benchmark.git
+      $ cd ocaml-monorepo-benchmark
+
+#. **Generate the duniverse Directory**: Use the ``generate-duniverse.sh``
+   script to create the ``duniverse`` directory at a desired location (e.g.,
+   ``/tmp``).
+
+   .. code:: console
+
+      $ ./generate-duniverse.sh /tmp
```

Comment:
Add a note that this will create a directory `/tmp/duniverse` containing all the packages in the monorepo.